Лаб. 5  6413 Чан Куанг Минь

quangminh-202 opened the pull request:
**ФИО:** Чан Куанг Минь
**Номер группы:** 6413
**Номер лабораторной:** 5
**Номер варианта:** 78
**Краткое описание предметной области:** Поликлиника
**Краткое описание добавленных фич:** Интеграция клиентского Blazor WebAssembly приложения с оркестратором .NET Aspire для системы управления поликлиникой. Разработан полнофункциональный веб-интерфейс с CRUD-операциями для врачей, пациентов и записей на приём, включающий аналитическую панель с визуализацией данных.

alxmcs commented on StoreApp/StoreApp.Server/Dto/CustomerGetDto.cs:
```diff
@@ -0,0 +1,18 @@
+﻿namespace StoreApp.Server.Dto;
+
+public class CustomerGetDto
```
Давай (чтобы я точно поверил, что курс разработки не прошел для тебя зря) переведем все dto с классов на record-ы

alxmcs commented on StoreApp/StoreApp.Server/MappingProfile.cs:
Добавить xml комментарии к коду

alxmcs commented on StoreApp/StoreApp.test/StoreApp.test.csproj:
Нарушение конвенций именования проектов

alxmcs commented on StoreApp/StoreApp.test/StoreTest.cs:
Не работают, кстати
![image](https://github.com/user-attachments/assets/27969288-78d8-45a3-94c4-a51496a2a502)

alxmcs commented on StoreApp/StoreApp.WebApplication/StoreApp.WebApplication.csproj:
Клиент не запускается, что достаточно очевидно, ведь у него есть референс серверного проекта
![image](https://github.com/user-attachments/assets/fa96af32-1e53-40e1-ab2a-ad4ef61cdd92)
![image](https://github.com/user-attachments/assets/423d0bdd-66ea-4edf-bb33-ec4cfa09374a)

alxmcs commented on StoreApp/StoreApp.WebApplication/appsettings.json:
Эти настройки в приложение не подтянутся, потому что они лежат не в той директории

alxmcs commented on StoreApp/StoreApp.WebApplication/Pages/Customer.razor:
```diff
@@ -0,0 +1,191 @@
+﻿@page "/customers"
+@using StoreApp.Server.Dto
+@inject HttpClient Http
```
Лишняя служба

alxmcs commented on StoreApp/StoreApp.WebApplication/Pages/Product.razor:
```diff
@@ -0,0 +1,198 @@
+﻿@page "/products"
+@inject HttpClient Http
```
Лишняя служба

alxmcs commented on StoreApp/StoreApp.WebApplication/Layout/MainLayout.razor:
```diff
@@ -0,0 +1,23 @@
+
+<div class="page">
+    <div class="sidebar">
+        <MudSnackbarProvider />
+        <NavMenu />
+        <MudThemeProvider />
+        <MudPopoverProvider />
+        <MudDialogProvider />
```
Это компоненты библиотеки MudBlazor

alxmcs commented on StoreApp/StoreApp.WebApplication/Program.cs:
А регистрируется библиотека компонентов Blazorise

alxmcs reviewed (changes requested):
1. Акшон для запуска юнит-тестов зафейлился ![image](https://github.com/user-attachments/assets/55522180-1455-4ebb-a29a-f8e97eeaa8a3)
2. Код с 1 по 3 лабу все еще написан Ярославом Хохриным

RYTpL commented:
исправил все, сервак работает, тесты я проходил еще в середине декабря, с тех пор много значений да и функций менял, тесты не подгонял
![7McEt6BVmL](https://github.com/user-attachments/assets/9ac9f1a9-0400-45a0-8a03-d283c91d7ddb)
![devenv_c8D3qehw5d](https://github.com/user-attachments/assets/99582228-32c1-4a70-b3d8-0fefb74d4b1d)
![chrome_kbQc0Gx8Hn](https://github.com/user-attachments/assets/0828d1b7-a2bf-46e3-8d3b-806c44f18474)

alxmcs commented on StoreApp/StoreApp.Client/StoreApp.Client.csproj:
Не помню, существовал ли этот проект в прошлое ревью
Смысла его существования я не наблюдаю

alxmcs reviewed (changes requested):
* Юнит тесты не работают. Аргумент "тесты я проходил еще в середине декабря, с тех пор много значений да и функций менял, тесты не подгонял" не принимается.
* Клиент тоже - теперь он не собирается (по достаточно очевидным причинам):
![image](https://github.com/user-attachments/assets/96e48607-218a-4eba-bb87-816b9823a3b5)

quangminh-202 committed:
add lab 1

quangminh-202 committed:
delete app

quangminh-202 committed:
fix name app

quangminh-202 committed:
fix error

quangminh-202 committed:
add documentation

quangminh-202 committed:
add dotnet

quangminh-202 committed:
add readme

quangminh-202 committed:
refactor split Models/Enums, add required, use primary constructors, fix PascalCase

quangminh-202 committed:
fix test 5

quangminh-202 committed:
remove unnecessary blank lines

quangminh-202 committed:
edit enum, add sumaries

quangminh-202 committed:
add lab 2

quangminh-202 committed:
add lab 2 all

quangminh-202 committed:
fix DoctorMemoryRepository.cs

quangminh-202 committed:
delete test projects

quangminh-202 committed:
add Polyclinic.Infrastructure.InMemory

quangminh-202 committed:
change name project

quangminh-202 committed:
add reference

quangminh-202 committed:
add reference 2

quangminh-202 committed:
fix error

quangminh-202 committed:
add summary for application.contracts

quangminh-202 committed:
add summary for Polyclinic.Infrastructure.InMemory

quangminh-202 committed:
add summary for Polyclinic.Infrastructure.InMemory

quangminh-202 committed:
Merge branch 'main' of https://github.com/quangminh-202/enterprise-development-minh

quangminh-202 committed:
delete server wrong

quangminh-202 committed:
create AutoMapper

quangminh-202 committed:
add interface service

quangminh-202 committed:
add CreateUpdateDto

quangminh-202 committed:
fix mapping

quangminh-202 committed:
add data and interfaces for domain

quangminh-202 committed:
fix controller

quangminh-202 committed:
fix api.host

quangminh-202 committed:
fix application

quangminh-202 committed:
fix infrastructure

quangminh-202 committed:
add .gitignore

quangminh-202 committed:
fix test

quangminh-202 committed:
fix test 4

quangminh-202 committed:
delete blank lines

quangminh-202 committed:
added logger

quangminh-202 committed:
add primary constructor and fix

quangminh-202 committed:
fix test

quangminh-202 committed:
add database class and migrations mongoDb

quangminh-202 committed:
change for connect to mongodb

quangminh-202 committed:
Fix: test use database

quangminh-202 committed:
Fix: test.yml

quangminh-202 committed:
test dotnet.yml

quangminh-202 committed:
fix test

quangminh-202 committed:
add ef core

quangminh-202 committed:
use primary constructor

quangminh-202 committed:
fix test

quangminh-202 committed:
delete blank line

quangminh-202 committed:
fix dto and warnings nullable

quangminh-202 committed:
fix test

quangminh-202 committed:
fix appointmentEfRepo

quangminh-202 committed:
try lab_4 first commit

quangminh-202 committed:
try lab 4

quangminh-202 committed:
add lab4 first

quangminh-202 committed:
fix nullable

quangminh-202 committed:
remove desktop.ini

quangminh-202 committed:
change readme

quangminh-202 committed:
Delete desktop.ini

quangminh-202 committed:
add nats.host and validator

quangminh-202 committed:
Merge branch 'lab_4' of https://github.com/quangminh-202/enterprise-development-minh into lab_4

quangminh-202 committed:
add branch github action lab_4

quangminh-202 committed:
fix test

quangminh-202 committed:
fix test: update expected values for AppointmentsCurrentMonthInSelectedRoom

quangminh-202 committed:
fix workflow: update solution file path from PolyclinicLab.sln to Polyclinic.sln

quangminh-202 committed:
trigger workflow

quangminh-202 committed:
add Polyclinic.sln to repository

quangminh-202 committed:
refactor: make ack parameter nullable and rename SendAppointmentsAsync to SendBatchAsync

quangminh-202 committed:
fix nullable

quangminh-202 committed:
feat: Complete Polyclinic system with Blazor WASM client and NATS integration

- Add Blazor WASM client with full CRUD operations for Doctors, Patients, Appointments
- Implement NATS JetStream integration with Generator and Validator services
- Add comprehensive analytics dashboard with multiple data visualizations
- Fix Generator service retry logic and improve error handling
- Add default sorting to all repositories (ID ascending order)
- Improve UI consistency across all DataGrid components with ID badges
- Update README with Russian documentation and complete setup instructions
- Clean up redundant code and optimize component structure

quangminh-202 committed:
fix test

quangminh-202 committed:
fix last test
